Sources/ManageServer.php

```diff
@@ -1544,28 +1544,28 @@ function ShowPHPinfoSettings()
  */
 function loadCacheAPIs()
 {
-	global $sourcedir, $txt;
+	global $sourcedir, $txt, $cacheAPIdir, $cacheAPIImplementationdir;
 
 	// Make sure our class is in session.
-	require_once($sourcedir . '/Class-CacheAPI.php');
+	require_once($cacheAPIdir . '/CacheAPI.php');
 
 	$apis = array();
-	if ($dh = opendir($sourcedir))
+	if ($dh = opendir($cacheAPIImplementationdir))
 	{
 		while (($file = readdir($dh)) !== false)
 		{
-			if (is_file($sourcedir . '/' . $file) && preg_match('~^CacheAPI-([A-Za-z\d_]+)\.php$~', $file, $matches))
+			if (is_file($sourcedir . '/' . $file))
 			{
-				$tryCache = strtolower($matches[1]);
-
 				require_once($sourcedir . '/' . $file);
-				$cache_class_name = $tryCache . '_cache';
-				$testAPI = new $cache_class_name();
+
+				$testAPI = new $file();
```

Comment:
How is this to work if $file is a file name and your new is trying to initialize an object?